x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/VerifierTests.java

```diff
@@ -2014,6 +2018,80 @@ public void testInvalidMapOption() {
         );
     }
 
+    public void testMatchOptions() {
+        assumeTrue("Match options are not available", EsqlCapabilities.Cap.MATCH_FUNCTION_OPTIONS.isEnabled());
+
+        // Check positive cases
+        query("FROM test | WHERE match(first_name, \"Jean\", {\"analyzer\": \"standard\"})");
+        query("FROM test | WHERE match(first_name, \"Jean\", {\"boost\": 2.1})");
+        query("FROM test | WHERE match(first_name, \"Jean\", {\"fuzziness\": 2})");
+        query("FROM test | WHERE match(first_name, \"Jean\", {\"fuzziness\": \"AUTO\"})");
+        query("FROM test | WHERE match(first_name, \"Jean\", {\"fuzzy_transpositions\": false})");
+        query("FROM test | WHERE match(first_name, \"Jean\", {\"lenient\": false})");
+        query("FROM test | WHERE match(first_name, \"Jean\", {\"max_expansions\": 10})");
+        query("FROM test | WHERE match(first_name, \"Jean\", {\"minimum_should_match\": \"2\"})");
+        query("FROM test | WHERE match(first_name, \"Jean\", {\"operator\": \"AND\"})");
+        query("FROM test | WHERE match(first_name, \"Jean\", {\"prefix_length\": 2})");
+        query("FROM test | WHERE match(first_name, \"Jean\", {\"auto_generate_synonyms_phrase_query\": true})");
+
+        // Check all data types for available options
+        DataType[] optionTypes = new DataType[] {
+            DataType.INTEGER,
+            DataType.LONG,
+            DataType.FLOAT,
+            DataType.DOUBLE,
+            DataType.KEYWORD,
+            DataType.BOOLEAN };
+        for (Map.Entry<String, DataType> allowedOptions : Match.ALLOWED_OPTIONS.entrySet()) {
+            String optionName = allowedOptions.getKey();
+            DataType optionType = allowedOptions.getValue();
+            for (DataType currentType : optionTypes) {
+                String optionValue = switch (currentType) {
+                    case BOOLEAN -> String.valueOf(randomBoolean());
+                    case INTEGER -> String.valueOf(randomIntBetween(0, 100000));
+                    case LONG -> String.valueOf(randomLong());
+                    case FLOAT -> String.valueOf(randomFloat());
+                    case DOUBLE -> String.valueOf(randomDouble());
+                    case KEYWORD -> randomAlphaOfLength(10);
+                    default -> throw new IllegalArgumentException("Unsupported option type: " + currentType);
+                };
+                String queryOptionValue = optionValue;
+                if (currentType == KEYWORD) {
+                    queryOptionValue = "\"" + optionValue + "\"";
+                }
+
+                String query = "FROM test | WHERE match(first_name, \"Jean\", {\"" + optionName + "\": " + queryOptionValue + "})";
+                // Check conversion is possible
+                try {
+                    DataTypeConverter.convert(optionValue, optionType);
+                    query(query);
+                } catch (InvalidArgumentException e) {
```

Comment:
Is this exception ever thrown? could we fix how we generate the values? this test would be more compelling without having this check here.

Reply:
This exception can be thrown. This is used to check if the option can be casted to the appropriate type, something that is done by `Match` to convert option types where possible.

Catching this exception allows us to iterate on all possible option types for completeness. Otherwise tests will be more verbose and won't cover all cases.

I'm happy to hear about alternatives

Reply:
I see - I think a comment here would not hurt
We can also choose to be a bit more explicit:

```
if (currentType == optionType) {
  // we expect the query to be valid
  query(query);
} else {
  // there is no guarantee that the generated value cannot be cast to the appropriate type
  // e.g `optionType` is `DOUBLE` and `optionValue` is an INTEGER
  // we need to handle both cases where the query is valid or fails with a casting error.
  try {
     DataTypeConverter.convert(optionValue, optionType);
     query(query);
   } ...
}
```

Reply:
Sure, added some comments in cd9a2c21309a68580df75ce715d90c9c7baf8db7